src/plugins/intel_cpu/src/nodes/executors/executor_factory.hpp

```diff
@@ -146,17 +146,27 @@ class ExecutorFactory {
      *
      */
     ExecutorPtr make(MemoryArgs& memory) {
-        const size_t implId = select(memory);
-        const auto& impl = m_suitableImplementations[implId].get();
-        if (m_implementationRequiresFallback[implId]) {
-            executor::Config<Attrs> config{memoryDescsFromMemory(memory), m_attrs, m_postOps};
-            if (auto fallbackConfig = impl.requiresFallback(config)) {
-                return fallback<Attrs, NodeT>(config, *fallbackConfig, memory, m_context, impl.name());
+        auto createExec = [this](MemoryArgs& memory, size_t implId) -> ExecutorPtr {
+            const auto& impl = m_suitableImplementations[implId].get();
+            if (m_implementationRequiresFallback[implId]) {
+                executor::Config<Attrs> config{memoryDescsFromMemory(memory), m_attrs, m_postOps};
+                if (auto fallbackConfig = impl.requiresFallback(config)) {
+                    return fallback<Attrs, NodeT>(config, *fallbackConfig, memory, m_context, impl.name());
+                }
             }
+            const auto executor = create(impl, memory, m_context);
+            if (!executor->update(memory)) {
+                return nullptr;
+            }
+            return executor;
+        };
+
+        auto implId = select(memory, 0);
+        auto executor = createExec(memory, implId);
+        while (!executor) {
+            implId = select(memory, ++implId);
```

Comment:
Cannot it be endless loop in case select cannot find suitable impl?

Reply:
Nope, the `implId` variable is always incremented, while the `select` call will throw if `implId >= m_suitableImplementations.size()`. Please refer to lines 235-239